terraform/modules/front_door/alarms.tf

```diff
@@ -0,0 +1,19 @@
+resource "aws_cloudwatch_metric_alarm" "unhealthy_hosts_count" {
+  alarm_actions             = [var.sns_topic_arn]
+  alarm_name                = "${var.prefix}-unhealthy-hosts-count"
+  comparison_operator       = "GreaterThanThreshold"
+  datapoints_to_alarm       = 1
+  evaluation_periods        = 1
+  metric_name               = "UnHealthyHostCount"
+  namespace                 = "AWS/ApplicationELB"
+  ok_actions                = [var.sns_topic_arn]
+  period                    = 60
+  statistic                 = "Maximum"
+  threshold                 = 0
+  insufficient_data_actions = []
+
+  dimensions = {
+    LoadBalancer = aws_lb.this.arn_suffix
+    TargetGroup  = aws_lb_target_group.this.arn_suffix
+  }
+}
```

Comment:
Do we also want an alarm for not enough healthy hosts? Or will that be sufficiently covered by the others?

Reply:
I think there's an additional state where one (or more) of the targets doesn't exist. E.g. if the service has one task running and it's healthy but there are supposed to be two, then that's still a problem even though there aren't any unhealthy hosts.

I guess that the attempts to start new tasks in that scenario ought to be being caught by the other alarms (my previous attempts at having something like this work in other scenarios was alarm for unhealthy and for too few healthy (and failed deployments), but didn't have anything looking at the ecs events).

Reply:
Ok fair enough, I misunderstood the potential states. I'll add that extra alarm now then 🙂

Reply:
Ok @RachaelBooth this is done now if you're happy to re-review. Unless I'm mistaken, the number of healthy hosts should equal the desired number of tasks, is that right? On that note I guess we will also need to move these alarms into a per-app module when we start the review apps work. I decided not to do that now because I thought it would make sense to do that at the same time we move the target group (as you described [here](https://dluhcdigital.atlassian.net/browse/CLDC-2811)) - especially since we don't know yet whether we're going to put the target group in the application module or a new module.